app/models/dao/TransactionsDao.scala

```diff
@@ -0,0 +1,52 @@
+package models.dao
+
+import javax.inject.Inject
+import models.{Contract, DBInput, DBOutput, DBTransaction, FullFilledTransaction}
+import models.database.TransactionsQueries._
+import models.service.{DBService, TransService}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TransactionsDao @Inject()(dbService: DBService, transService: TransService)(implicit ec: ExecutionContext) {
+
+  def transactionsByBlock(id: String): Future[List[DBTransaction]] = dbService.runAsync(getTransactionsByBlockId(id))
+
+  def inputsByTransaction(id: String): Future[List[DBInput]] = dbService.runAsync(getTransactionInputs(id))
+
+  def outputsByTransaction(id: String): Future[List[DBOutput]] = dbService.runAsync(getTransactionOutputs(id))
+
+  def transactionById(id: String): Future[Option[DBTransaction]] = {
+    transService.getTransactionById(id).flatMap { txOpt =>
+      if(txOpt.isEmpty) dbService.runAsync(getTransactionById(id))
+      else Future(txOpt)
```

Comment:
Here `txOpt` value has already completed. So you should use `Future.successful()`.